src/models/dirac_mixture_single_publisher_model.py

```diff
@@ -315,3 +316,173 @@ def predict(
             for c in self.components
         ]
         return sum([w * pmf for w, pmf in zip(self.ws, scaled_component_pmfs)])
+
+
+class DiracMixtureSinglePublisherModel(ReachCurve):
+    def __init__(
+        self,
+        data: List[ReachPoint],
+        ncomponents: int = 200,
+    ):
+        """Constructs a Dirac mixture single publisher model.
+
+        Args:
+            data:  A list consisting of the single ReachPoint to which the model
+                is to be fit.
+            ncomponents:  Number of components in the Poisson mixture.  Based on
+                our experience, 100 components are enough, and more components
+                are also fine (they will not introduce overfitting).
+        """
+        if len(data) != 1:
+            raise ValueError("Exactly one ReachPoint must be specified")
+        if data[0].impressions[0] < 0.001:
+            raise ValueError("Attempt to create model with 0 impressions")
+        if data[0].impressions[0] < data[0].reach(1):
+            raise ValueError(
+                "Cannot have a model with fewer impressions than reached people"
+            )
+        self._data = data
+        self._reach_point = data[0]
+        if data[0].spends:
+            self._cpi = data[0].spends[0] / data[0].impressions[0]
+        else:
+            self._cpi = None
+        if self._reach_point.universe_size is None:
+            raise ValueError(
+                "Dirac mixture model only works when the universe size is given"
```

Comment:
I suggest a more informative message, for example: 
"A fit requires the universe size to be known, please provide a ReachPoint with a known universe size instead."

Reply:
Well said!!